xdsl/transforms/convert_stencil_to_csl_stencil.py

```diff
@@ -183,50 +186,155 @@ class ConvertSwapToPrefetchPattern(RewritePattern):
     Translates dmp.swap to csl_stencil.prefetch
     """
 
-    @op_type_rewrite_pattern
-    def match_and_rewrite(self, op: dmp.SwapOp, rewriter: PatternRewriter, /):
+    def untensorized_match_and_rewrite(
+        self, op: dmp.SwapOp, rewriter: PatternRewriter, /
+    ):
         # remove op if it contains no swaps
         if len(op.swaps) == 0:
             rewriter.erase_matched_op(False)
             return
 
-        assert all(
-            len(swap.size) == 3 for swap in op.swaps
-        ), "currently only 3-dimensional stencils are supported"
-
-        assert all(
-            swap.size[:2] == (1, 1) for swap in op.swaps
-        ), "invoke dmp to decompose from (x,y,z) to (1,1,z)"
-
-        # check that size is uniform
-        uniform_size = op.swaps.data[0].size[2]
-        assert all(
-            swap.size[2] == uniform_size for swap in op.swaps
-        ), "all swaps need to be of uniform size"
+        # uses have to be retrieved *before* the loop because of the rewriting happening inside the loop
```

Comment:
This comment is somewhat misleading, since there is no rewriting happening in this loop